eodag/rest/utils/rfc3339.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# MIT License
+
+# Copyright (c) 2020 Arturo AI
+# Copyright (c) 2023 CS Systemes d'Information, https://www.csgroup.eu/
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+rfc3339.
+
+Taken from stac_fastapi.types to not depend of stac_fastapi.types until it supports pydantic v2
+https://github.com/stac-utils/stac-fastapi/issues/593
+
+"""
+import re
+from datetime import datetime, timezone
+from typing import Optional, Tuple
+
+import iso8601
```

Comment:
please use already in deps https://dateutil.readthedocs.io/en/latest/parser.html#dateutil.parser.isoparse instead of adding a new dependency